app/join/step2/BottomModal.tsx

```diff
@@ -0,0 +1,93 @@
+'use client';
+import { useEffect, useState } from 'react';
+
+import Button from '@/components/common/Button';
+import CircleCheckbox from '@/components/common/Checkbox/CircleCheckbox';
+import BottomUpModal from '@/components/common/Modal/BottomUpModal';
+import { useModal } from '@/hooks/useModal';
+
+type AgreeCheckListType = {
+  name: string;
+  required: boolean;
+  isAgreed: boolean;
+};
+
+export default function BottomModal() {
+  const { isModalOpen, openModal, closeModal } = useModal<'modal'>();
+  const [agreeCheckList, setAgreeCheckList] = useState<AgreeCheckListType[]>([
+    {
+      name: '서비스 이용약관 동의',
+      required: true,
+      isAgreed: false,
+    },
+    {
+      name: '개인정보 수집 및 이용 동의',
+      required: true,
+      isAgreed: false,
+    },
+  ]);
+
+  useEffect(() => {
+    openModal('modal');
+  }, [openModal]);
+
+  return (
+    <BottomUpModal
+      snap={300}
+      isModalOpen={isModalOpen}
+      onClose={closeModal}
+      disableDrag
+      text={<div className="text-center font-700">약관 동의</div>}
+    >
+      <section>
+        <CircleCheckbox
+          text="전체 동의"
+          checked={agreeCheckList.every((agree) => agree.isAgreed)}
+          onClick={() => {
+            if (agreeCheckList.every((agree) => agree.isAgreed)) {
+              setAgreeCheckList((agreeCheckList) =>
+                agreeCheckList.map((agree) => ({ ...agree, isAgreed: false }))
+              );
+            } else {
+              setAgreeCheckList((agreeCheckList) =>
+                agreeCheckList.map((agree) => ({ ...agree, isAgreed: true }))
+              );
+            }
+          }}
```

Comment:
이벤트 핸들러가 길어지면 함수로 빼서 관리하는게 좋을 것 같아요!

근데 이 부분은 이렇게 바꿀 수 있을 것 같네용

```suggestion
          onClick={() => {
            setAgreeCheckList((agreeCheckList) =>
              agreeCheckList.map((agree) => ({
                ...agree,
                isAgreed: !agreeCheckList.every((agree) => agree.isAgreed),
              }))
            );
          }}
```

Reply:
호 함수 하나로 관리 할 수 있군요 ! 

좋아요 함수가 기니, 가독성을 위해서 위에 따로 뺴둘게요 👍

Reply:
이부분 수정 아직 안된것 같아요! 수정 부탁드립니당

Reply:
push를 안했네요, push 완료했습니다 😄